### PR TITLE
feat: adding RNSM to boss room [MTT-3267]

### PR DESCRIPTION
**Pull request by SamuelBellomo**
### Description

Basic RNSM, with some inputs to toggle it. We might need to tweak it in the future depending on our needs, but the default prefab is there.

### Issue Number(s)

[MTT-3267](https://jira.unity3d.com/browse/MTT-3267)

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] JIRA ticket ID is in the PR title or at least one commit message
 - [ ] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

**Comment by SamuelBellomo** [comment]
This is part 3
This shouldn't be merged until RNSM is packagified
To run this, you should make sure to include RNSM in your manifest.json

**Comment by SamuelBellomo** [comment]
RNSM package to add
    "com.unity.multiplayer.tools": "https://github.com/Unity-Technologies/com.unity.multiplayer.tools.git#develop",

**Review by pdeschain — APPROVED** [review]

File: Packages/com.unity.multiplayer.samples.coop/Utilities/DontDestroyOnLoad.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+using System;
+using UnityEngine;
+
+public class DontDestroyOnLoad : MonoBehaviour
```
Comment (fernando-cortez) [inline Packages/com.unity.multiplayer.samples.coop/Utilities/DontDestroyOnLoad.cs:4]: Namespace nitpick.

File: Packages/com.unity.multiplayer.samples.coop/Utilities/Net/RNSM/AutoHide.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+using System.Collections;
+using UnityEngine;
+
+public class AutoHide : MonoBehaviour
```
Comment (fernando-cortez) [inline Packages/com.unity.multiplayer.samples.coop/Utilities/Net/RNSM/AutoHide.cs:4]: Namespace nitpick.

**Review by fernando-cortez — APPROVED** [review]
Besides nitpicks, gtg.

File: Packages/com.unity.multiplayer.samples.coop/Utilities/DontDestroyOnLoad.cs (hunk 1 above)
Comment (SamuelBellomo) [reply Packages/com.unity.multiplayer.samples.coop/Utilities/DontDestroyOnLoad.cs]: fixed

File: Packages/com.unity.multiplayer.samples.coop/Utilities/Net/RNSM/AutoHide.cs (hunk 2 above)
Comment (SamuelBellomo) [reply Packages/com.unity.multiplayer.samples.coop/Utilities/Net/RNSM/AutoHide.cs]: fixed

**Review by LPLafontaineB — APPROVED** [review]

**Review by fernando-cortez — APPROVED** [review]

**Pull request merged**